src/techui_builder/generate.py

```diff
@@ -157,6 +214,68 @@ def _create_widget(self, component: Entry) -> EmbeddedDisplay | ActionButton:
 
         return new_widget
 
+    def layout_widgets(self, widgets: list[EmbeddedDisplay | ActionButton]):
+        group_spacing = 30
+        max_group_height = 800
+        spacing_x = 20
+        spacing_y = 30
+        # Group tiles by size
+        groups = defaultdict(list)
+        for widget in widgets:
+            key = self._get_screen_dimensions_object(widget)
+
+            groups[key].append(widget)
+
+        # Sort groups by width (optional)
+        sorted_widgets = []
+        sorted_groups = sorted(groups.items(), key=lambda g: g[0][0], reverse=True)
+        current_x = 0
+        current_y = 0
+        column_width = 0
+        column_levels = []
+
+        for (h, w), group in sorted_groups:
+            for widget in group:
+                placed = False
+                for level in column_levels:
+                    level_y, level_x = self._get_screen_position_object(level[0])
```

Comment:
`level_x` is currently not used. Is it needed? If not, just call it `_`
```suggestion
                    level_y, _ = self._get_screen_position_object(level[0])
```